### PR TITLE
[Concurrency] Revise `Async-` related files doc

### PR DESCRIPTION
**Pull request by Jager-yoo**
- Revise `// Prints` comments style like the other stdlib files.
- Remove verbose string interpolations and extra spaces to be read more clearly.
  - (e.g.)
  ```diff
  - print ("\(error)")
  + print(error)
  ```
- Remove some unneeded parentheses.
  - (e.g.)
  ```diff
  - if (random % 5 == 0) { ... }
  + if random % 5 == 0 { ... }
  ```
- Replace the majority of ' : ' with ': '
  - (ref.) #26218
- Fix wrong indentation
- Keep files with a single empty line at the end
  - (ref.) SwiftLint Rules - ['Trailing Newline'](https://realm.github.io/SwiftLint/trailing_newline.html) and other stdlib files.

### CODE REVIEW

**Comment by Jager-yoo** [comment]
Hi, @invalidname!
Could you please review my change proposals?
Cause most of the changes are related to your previous PR. #37383

**Comment by phausler** [comment]
Overall the changes look sound to me. I defer to the folks doing documentation publishing for style guidance.

**Review by phausler — APPROVED** [review]

**Comment by Jager-yoo** [comment]
Really appreciate it! 👍🏻

**Comment by Jager-yoo** [comment]
Hi, @lorentey!
Could you check doc styles and ask @swift-ci to smoke test this?

File: stdlib/public/Concurrency/AsyncStream.swift [hunk 1]
```diff
@@ -319,15 +319,13 @@ public struct AsyncStream<Element> {
   /// the `unfolding` parameter label.
   ///
   ///     let stream = AsyncStream<Int> {
-  ///             await Task.sleep(1 * 1_000_000_000)
-  ///             return Int.random(in: 1...10)
-  ///         }
-  ///         onCancel: { @Sendable () in print ("Canceled.") }
-  ///     )
```
Comment (invalidname) [inline stdlib/public/Concurrency/AsyncStream.swift:326]: Nice job catching this stray `)` that shouldn't be there.

File: stdlib/public/Concurrency/AsyncDropFirstSequence.swift [hunk 2]
```diff
@@ -22,12 +22,12 @@ extension AsyncSequence {
   ///
   /// In this example, an asynchronous sequence called `Counter` produces `Int`
   /// values from `1` to `10`. The `dropFirst(_:)` method causes the modified
-  /// sequence to ignore the values `0` through `4`, and instead emit `5` through `10`:
+  /// sequence to ignore the values `1` through `3`, and instead emit `4` through `10`:
```
Comment (invalidname) [inline stdlib/public/Concurrency/AsyncDropFirstSequence.swift:25]: Right. This way matches the actual output on line 30.

**Review by invalidname — APPROVED** [review]
Looks good. Thanks for these corrections.

File: stdlib/public/Concurrency/AsyncStream.swift (hunk 1 above)
Comment (Jager-yoo) [reply stdlib/public/Concurrency/AsyncStream.swift:326]: Thanks! 😄 
Note that I also edited the word "Canceled." → `"Cancelled."` (double `l`)

**Comment by Jager-yoo** [comment]
Thanks for reviewing, @invalidname!
Could you ask @swift-ci to smoke test and merge this? Or should I ask someone else?

File: stdlib/public/Concurrency/AsyncStream.swift (hunk 1 above)
Comment (Jager-yoo) [reply stdlib/public/Concurrency/AsyncStream.swift:326]: @amartini51 Thanks for following up!
The [Apple Style Guide - C](https://help.apple.com/applestyleguide/#/apsgb744e4a3#apdb51b39ece6204) definitely says that we should use one `l` for `"canceled"`.

On [lines 121-122](https://github.com/apple/swift/blob/80ef80fef9a891c39a7b9a119041b5c9f0e01e21/stdlib/public/Concurrency/AsyncStream.swift#L121-L122) in the very same file, however, there is `case cancelled` in use.

```swift
/// The stream finished as a result of cancellation.
case cancelled
```

Actually, "cancelled" is being used much more than "canceled" in overall Swift repo. 🧐
Would it be better to just revert this word for now? How do you think?

Comment (amartini51) [reply stdlib/public/Concurrency/AsyncStream.swift:326]: Let's revert this change please, so we're not making the problem worse.  Someone (possibly me or @invalidname) can take another pass over the file to correct the other "cancelled" spellings in a different PR.

File: stdlib/public/Concurrency/AsyncStream.swift [hunk 3]
```diff
@@ -319,15 +319,13 @@ public struct AsyncStream<Element> {
...
+  ///         await Task.sleep(1 * 1_000_000_000)
+  ///         return Int.random(in: 1...10)
+  ///     } onCancel: { @Sendable () in print("Cancelled.") }
```
Comment (amartini51) [inline stdlib/public/Concurrency/AsyncStream.swift:324]:
```suggestion
  ///     } onCancel: { @Sendable () in print("Canceled.") }
```

**Comment by amartini51** [comment]
@swift-ci Please smoke test.

File: stdlib/public/Concurrency/AsyncStream.swift (hunk 1 above)
Comment (amartini51) [reply stdlib/public/Concurrency/AsyncStream.swift:326]: Regarding lines 121-121, `case cancelled` is an example of the Cocoa naming guidelines I mentioned.  You can see this style difference in API reference for `Task` and and related types — symbols with "cancelled" in its name have "canceled" in their docs.

Both style guides use the spelling "cancellation".

Comment (Jager-yoo) [reply stdlib/public/Concurrency/AsyncStream.swift:326]: Ohh, now I see. Documentation follows "canceled", while the API naming follows "cancelled".
Sorry for additional revert commit, and thank you so much for being part of this.

Comment (amartini51) [reply stdlib/public/Concurrency/AsyncStream.swift:326]: Reading my original post, I could have been clearer that there were two style guides involved here, one for code and one for docs. Sorry about that. Thanks for the fixes in this PR!

**Pull request merged**